src/Swarm/TUI/Controller.hs

```diff
@@ -197,7 +199,7 @@ runGameTick = zoom gameState gameTick
 
 -- | Update the UI.  This function is used after running the
 --   game for some number of ticks.
-updateUI :: StateT AppState (EventM Name) ()
+updateUI :: StateT AppState (EventM Name) Bool
 updateUI = do
 
   loadVisibleRegion
```

Comment:
not sure what to do with the result of `loadVisibleRegion`, it seems like it always modifies `gameState . world`

Reply:
`loadVisibleRegion` just makes sure that the world has loaded all tiles that overlap the visible area into its tile cache.  It doesn't affect whether the screen needs to be redrawn.